src/systems/filecoin_markets/storage_market/storage_market_actor_state.go

```diff
@@ -219,14 +265,11 @@ func (st *StorageMarketActorState_I) _assertActiveDealState(rt Runtime, dealID d
 	}
 }
 
-func (st *StorageMarketActorState_I) _safeGetBalance(rt Runtime, participant addr.Address) StorageParticipantBalance {
-	balance, found := st.Balances()[participant]
-
-	if !found {
-		rt.AbortStateMsg("sma._safeGetBalance: participant balance not found.")
-	}
-
-	return balance
+func (st *StorageMarketActorState_I) _addressEntryExists(address addr.Address) bool {
+	_, foundEscrow := actor.BalanceTable_GetEntry(st.EscrowTable(), address)
+	_, foundLocked := actor.BalanceTable_GetEntry(st.LockedReqTable(), address)
+	Assert(foundEscrow == foundLocked)
```

Comment:
I dont think this check is correct, foundEscrow >= foundLocked? Equality check means that withdraw is called whenever unlock is called.

Reply:
foundEscrow and foundLocked are booleans indicating whether the address exists -- the check is just asserting the invariant that the two tables are consistent (w/r/to which addresses exist in them)

Reply:
add a comment?